test/test_rpc_endpoints.py

```diff
@@ -0,0 +1,641 @@
+    invoke_info = json.loads(resp.data.decode("utf-8"))
+    return invoke_info
+
+
+def pad_zero(felt: str) -> str:
+    """
+    Convert felt with format `0xValue` to format `0x0Value`
+    """
+    felt = felt.lstrip("0x")
+    return "0x0" + felt
+
+
+# pylint: disable=unused-argument
+def test_get_block_by_number(deploy_info):
+    """
+    Get block by number
+    """
+    resp = rpc_call(
+        "starknet_getBlockByNumber", params={"block_number": 0}
+    )
+    block = resp["result"]
+    expected = {
+        "parent_hash": "0x0000000000000000000000000000000000000000000000000000000000000000",
+        "block_number": 0,
+        "status": "ACCEPTED_ON_L2",
+        "sequencer": "0x0000000000000000000000000000000000000000",
+        "old_root": "0x0",
+    }
+
+    for key, value in expected.items():
+        assert block[key] == value
+
+
+def test_get_block_by_number_raises_on_incorrect_number(deploy_info):
+    """
+    Get block by incorrect number
+    """
+    ex = rpc_call(
+        "starknet_getBlockByNumber", params={"block_number": 1234}
+    )
+
+    assert ex["error"] == {
+        "code": 26,
+        "message": "Invalid block number"
+    }
+
+
+def test_get_block_by_hash(deploy_info):
+    """
+    Get block by hash
+    """
+    block_hash = gateway_call("get_block", blockNumber=0)["block_hash"]
+
+    resp = rpc_call(
+        "starknet_getBlockByHash", params={"block_hash": block_hash}
+    )
+    block2 = resp["result"]
+    expected = {
+        "block_hash": block_hash,
+        "parent_hash": "0x0000000000000000000000000000000000000000000000000000000000000000",
+        "block_number": 0,
+        "status": "ACCEPTED_ON_L2",
+        "sequencer": "0x0000000000000000000000000000000000000000",
+        "old_root": "0x0",
+    }
+
+    for key, value in expected.items():
+        assert block2[key] == value
+
+
+def test_get_block_by_hash_raises_on_incorrect_hash(deploy_info):
+    """
+    Get block by incorrect hash
+    """
+    ex = rpc_call(
+        "starknet_getBlockByHash", params={"block_hash": "0x0"}
+    )
+
+    assert ex["error"] == {
+        "code": 24,
+        "message": "Invalid block hash"
+    }
+
+
+def test_get_storage_at(deploy_info):
+    """
+    Get storage at address
+    """
+    block = gateway_call("get_block", blockNumber=0)
+
+    contract_address: str = deploy_info["address"]
+    key: str = str(hex(get_storage_var_address("balance")))
+    block_hash: str = block["block_hash"]
+
+    resp = rpc_call(
+        "starknet_getStorageAt", params={
+            "contract_address": contract_address,
+            "key": key,
+            "block_hash": block_hash,
+        }
+    )
+    storage = resp["result"]
+
+    assert storage == "0x0"
+
+
+def test_get_storage_at_raises_on_incorrect_contract(deploy_info):
+    """
+    Get storage at incorrect contract
+    """
+    block = gateway_call("get_block", blockNumber=0)
+
+    key: str = str(hex(get_storage_var_address("balance")))
```

Comment:
No need for applying `str` on `hex` as `hex` already returns a string. In other places as well.

Reply:
Oops